src/lib/storage.js

```diff
@@ -51,6 +56,17 @@ class Storage extends ScratchStorage {
     getAssetGetConfig (asset) {
         return `${this.assetHost}/internalapi/asset/${asset.assetId}.${asset.dataFormat}/get/`;
     }
+    getAssetCreateConfig (asset) {
```

Comment:
I appreciate the comment below (and the one above) but I find them hard to understand. "There is no such thing" meaning, there's no API call for it? But it does exist, as you say in the same sentence? Sorry, I know this stuff is hard to write, I'm just not following!

Reply:
Since an asset is stored as the MD5 of its contents, there's no such thing as "updating" an asset, because if the contents were to change, the filename would change, and then you're not storing that asset anymore, you're creating a new one with a different MD5. So there's literally no "update" route for assets — if you `PUT` to the assets server, you'll get a `404` response.

But, the way storage is written, it tries to decide which way to store based on the presence of an assetId. If it has an ID, then it assumes you want to update the asset with that ID, not store a new one.  This is, once again, a shortcoming of the way storage is designed w/r/t web storage.  So this is a hack to get around that.

I don't know how to put that more concisely though, hence the wordy and confusing comment... any suggestions?

Reply:
Sorry, still don't understand. 

* "Stored as the MD5 of its contents" -- you mean, the MD5 is used as the key name for the file? Or something like that?
* "Which way to store" -- wait, there's another way besides that? Now there's two ways to store that I can't describe...
* "update the asset with that ID, not store a new one" -- here's updating again, which doesn't exist! I'm super lost :( What's the role of the assetId here? Is that different from the MD5?

Reply:
Yes exactly, the MD5 is the key name on S3 (actually MD5.<data type>, e.g. MD5.svg or MD5.wav)

Maybe also confusing is when talking about assets in general in the context of scratch-storage, I don't mean costumes and sounds, I mean the objects that scratch-storage interacts with, which may also be project sb3 files, project json files, etc. Anything that needs to be stored: https://github.com/LLK/scratch-storage/blob/develop/src/Asset.js and each Asset has a type: https://github.com/LLK/scratch-storage/blob/develop/src/AssetType.js

"Which way to store" — there are two ways to store in general, create or update (POST or PUT on the web). Since scratch-storage is a general tool, it assumes for any asset you will have a config for doing both of these things. It's not set up specifically for the assets server, we have to do that (that's what I'm doing in this PR).

> "update the asset with that ID, not store a new one" -- here's updating again, which doesn't exist! I'm super lost :( What's the role of the assetId here? Is that different from the MD5?

I tried to write this out a bunch of times but I don't think I can concisely describe everything that happens when you store something in this comment. Please take a look again at https://github.com/LLK/scratch-storage/blob/develop/src/WebHelper.js so you understand what `WebHelper.store` does, and after that let's chat offline to figure out what comment would be helpful here.